src/build/build.ts

```diff
@@ -77,9 +75,9 @@ export function build(options: BuildOptions, config: ProjectConfig): Promise<any
     logger.debug(`Reading source files...`);
     let sourcesStream = polymerProject.sources()
       .pipe(polymerProject.splitHtml())
-      .pipe(gulpif(/\.js$/, uglify(optimizeOptions.js)))
-      .pipe(gulpif(/\.css$/, cssSlam(optimizeOptions.css)))
-      .pipe(gulpif(/\.html$/, htmlmin(optimizeOptions.html)))
+      .pipe(gulpif(/\.js$/, new JSOptimizeStream(optimizeOptions.js)))
```

Comment:
@justinfagnani Check out the new `optimize-streams.ts` and let me know what you think. Personally I prefer the extra code vs. inheritance here, but this definitely does have better code reuse.